src/rdkafka_cgrp.c

```diff
@@ -3738,17 +3756,40 @@ rd_kafka_consumer_group_metadata_new (const char *group_id) {
 
         cgmetadata = rd_calloc(1, sizeof(*cgmetadata));
         cgmetadata->group_id = rd_strdup(group_id);
+        cgmetadata->generation_id = -1;
+
+        return cgmetadata;
+}
+
+rd_kafka_consumer_group_metadata_t *
+rd_kafka_consumer_group_metadata_new2 (const char *group_id,
```

Comment:
Suggest changing group_metadata_new() to simply call this function to avoid code duplication.

Reply:
haha, i forgot i wasn't in a constructor